index.js

```diff
@@ -175,6 +189,10 @@ Filter.prototype.build = function() {
       return result;
     }));
   }).then(function(result) {
+    return Promise.all(pendingWork).then(function() {
```

Comment:
I updated this to use RSVP.allSettled, and added a test to make sure this is happening correctly